### PR TITLE
hotfix release should check all approved revisions

### PR DESCRIPTION
**Pull request by praveenkuttappan**
Interim solution until we have package version available for all languages to compare  within semver ranges. Currently this check will only be done when package is ready for release.

### CODE REVIEW

**Comment by praveenkuttappan** [comment]
@pakrym : Can you please have a look at this PR when you get a chance?

File: src/dotnet/APIView/APIViewWeb/Repositories/ReviewManager.cs [hunk 1]
```diff
@@ -359,15 +359,16 @@ private async Task<bool> IsReviewSame(ReviewRevisionModel revision, RenderedCode
             return lastRevisionTextLines.SequenceEqual(fileTextLines);
         }
 
-        public async Task<ReviewModel> CreateMasterReviewAsync(ClaimsPrincipal user, string originalName, string label, Stream fileStream, bool runAnalysis)
+        public async Task<ReviewRevisionModel> CreateMasterReviewAsync(ClaimsPrincipal user, string originalName, string label, Stream fileStream, bool compareAllRevisions)
```
Comment (pakrym) [inline src/dotnet/APIView/APIViewWeb/Repositories/ReviewManager.cs:362]: Is this only used for auto-reviews?

File: src/dotnet/APIView/APIViewWeb/Repositories/ReviewManager.cs [hunk 2]
```diff
@@ -380,10 +381,23 @@ public async Task<ReviewModel> CreateMasterReviewAsync(ClaimsPrincipal user, str
...
+                    foreach (var approvedRevision in review.Revisions.Where(r => r.IsApproved).Reverse())
+                    {
+                        if (await IsReviewSame(approvedRevision, renderedCodeFile))
+                        {
+                            return approvedRevision;
+                        }
+                    }
+                }
+
+                if (await IsReviewSame(lastRevision, renderedCodeFile))
```
Comment (pakrym) [inline src/dotnet/APIView/APIViewWeb/Repositories/ReviewManager.cs:398]: else?

**Review by pakrym — APPROVED** [review]

File: src/dotnet/APIView/APIViewWeb/Repositories/ReviewManager.cs (hunk 1 above)
Comment (praveenkuttappan) [reply src/dotnet/APIView/APIViewWeb/Repositories/ReviewManager.cs:362]: yes

File: src/dotnet/APIView/APIViewWeb/Repositories/ReviewManager.cs (hunk 2 above)
Comment (praveenkuttappan) [reply src/dotnet/APIView/APIViewWeb/Repositories/ReviewManager.cs:398]: createNewRevision is set to true by default to ensure new revision is created if reviews are not same.

**Review by chidozieononiwu — APPROVED** [review]

**Pull request merged**